frontend/public/components/instantiate-template.tsx

```diff
@@ -129,6 +141,69 @@ const stateToProps = (state: RootState) => ({
   models: state.k8s.getIn(['RESOURCES', 'models']),
 });
 
+const TemplateFormField: React.FC<TemplateFormFieldProps> = ({
+  name,
+  value,
+  displayName,
+  description,
+  onChange,
+  placeholder,
+  required,
+  helpID,
+}) => {
+  const [isTextArea, setIsTextArea] = React.useState(false);
+  const { t } = useTranslation('public');
+
+  const commonFormProps = {
+    'aria-describedby': helpID,
+    placeholder,
+    isRequired: required,
+    onChange,
+    value,
+    id: name,
+    name,
+  };
+
+  const toggleTooltipText = isTextArea
+    ? t('Compress to a single line of content. This may strip any new lines you have entered.')
+    : t(
+        'Expand to enter multiple lines of content. This is required if you need to include newline characters.',
+      );
+
+  const InputToggle = (
+    <Tooltip aria="none" aria-live="polite" content={toggleTooltipText}>
+      <button
+        aria-label={toggleTooltipText}
+        onClick={() => setIsTextArea(!isTextArea)}
+        type="button"
+        className={formStyles.formGroupLabelHelp}
+      >
+        {isTextArea ? <CompressIcon /> : <ExpandIcon />}
+      </button>
+    </Tooltip>
```

Comment:
Why another button is required? Can it be a TextArea for all the fields? Instead of TextInput we can have TextArea so that another button and another click can be reduced?

Reply:
The previous design included the toggle switch #13317, and when copying content which should be single line, if only textarea is rendered users may mistakenly include newline characters which could break deployments